### PR TITLE
Fmt std core

### PR DESCRIPTION
**Pull request by crodas**
## Description

Run`forc fmt` on the std-core

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-lib-core/src/primitives.sw [hunk 1]
```diff
@@ -70,8 +70,8 @@ impl u64 {
     /// fn foo() {
     ///     let val = u64::min();
     ///     assert(val == 0);
-    // }
     /// ```
+    // }
```
Comment (sdankel) [inline sway-lib-core/src/primitives.sw:74]: This looks like a bug

File: swayfmt/src/utils/map/newline.rs [hunk 2]
```diff
@@ -45,59 +39,50 @@ fn is_new_line_in_rope(rope: &Rope, index: usize) -> bool {
...
     let mut in_sequence = false;
     let mut sequence_start = 0;
-    let os_new_line = NEW_LINE.chars().collect::<Vec<_>>();
-    while let Some((char_index, char)) = input_iter.next() {
-        let next_char = input_iter.peek().map(|input| input.1);
-        if matches!(char, '/' | ';' | '}')
-            && is_new_line_next_in_iter(&mut input_iter, &os_new_line)
-        {
+    let mut bytes_offset = 0;
+    while let Some(char) = input_iter.next() {
```
Comment (sdankel) [inline swayfmt/src/utils/map/newline.rs:51]: Could we put changes to the formatter in a separate PR so it's clear what bug it's fixing?

**Review by sdankel — CHANGES_REQUESTED** [review]

File: sway-lib-core/src/primitives.sw [hunk 3]
```diff
@@ -13,8 +13,8 @@ impl u256 {
     /// fn foo() {
     ///     let val = u256::min();
     ///     assert(val == 0x0000000000000000000000000000000000000000000000000000000000000000u256);
-    // }
     /// ```
+    // }
```
Comment (JoshuaBatty) [inline sway-lib-core/src/primitives.sw:17]: same here as what sophie mentions below

File: sway-lib-core/src/raw_ptr.sw [hunk 4]
```diff
@@ -207,23 +207,23 @@ impl raw_ptr {
     /// ```
     pub fn read_byte(self) -> u8 {
         asm(r1: self, r2) {
-            lb r2 r1 i0;
+            lb   r2 r1 i0;
```
Comment (JoshuaBatty) [inline sway-lib-core/src/raw_ptr.sw:210]: I'm not sure I understand what the purpose of inserting 2 white spaces here is?

File: sway-lib-core/src/primitive_conversions.sw [hunk 5]
```diff
@@ -18,7 +18,7 @@ impl u64 {
     /// ```
     pub fn as_u256(self) -> u256 {
         let input = (0u64, 0u64, 0u64, self);
-        asm(input: input) {
+        asm( input: input ) {
```
Comment (sdankel) [inline sway-lib-core/src/primitive_conversions.sw:21]: The spacing around asm arguments looks odd. Was that intentional?

Comment (JoshuaBatty) [reply sway-lib-core/src/primitive_conversions.sw:21]: Are we able to just skip asm blocks in the formatter for now as we discussed during the sync yesterday?

File: swayfmt/src/utils/language/expr/mod.rs [hunk 6]
```diff
@@ -544,46 +550,110 @@ impl Format for Expr {
...
-                        write!(
-                            formatted_code,
-                            "\n{}{} ",
-                            formatter.indent_to_str()?,
-                            double_pipe_token.span().as_str()
-                        )?;
-                    }
-                    _ => {
-                        write!(formatted_code, " {} ", double_pipe_token.span().as_str())?;
+                let mut rhs_code = FormattedCode::new();
```
Comment (JoshuaBatty) [inline swayfmt/src/utils/language/expr/mod.rs:610]: This block of code looks very similar to the addition above. Can we abstract some of it away to reduce duplication?

**Review by JoshuaBatty — APPROVED** [review]

File: sway-lib-core/src/primitive_conversions.sw (hunk 5 above)
Comment (JoshuaBatty) [reply sway-lib-core/src/primitive_conversions.sw]: asm blocks looks fine now, marking this as resolved.

**Review by sdankel — APPROVED** [review]
